modules/compiler/riscv/source/riscv_pass_machinery.cpp

```diff
@@ -223,6 +223,11 @@ llvm::ModulePassManager RiscvPassMachinery::getLateTargetPasses() {
   // functions will be cloned to give a version using degenerate sub-groups and
   // a version using non-degenerate sub-groups, for selection by the runtime.
   tuner.degenerate_sub_groups = true;
+
+  // We run the handle barriers pass, which deals with work group reductions and
+  // broadcasts itself.
+  tuner.handling_barriers = true;
```

Comment:
I don't see how this all fits together. If `RefSi G1 WIPT` is relying on `handling_barriers = false` as the default value, does that mean that `RefSi M1` is doing that too? Because that would be incorrect.

Reply:
it would still be correct. replacing the reductions and broadcasts should still work in any case. but i don't know how those things worked so i made the changes in the targets i understood.

Reply:
well `true` is default now so i can resolve this..